mypy/checkstrformat.py

```diff
@@ -176,7 +176,7 @@ def check_mapping_str_interpolation(self, specifiers: List[ConversionSpecifier],
                                        'placeholder with key \'%s\' has type' % specifier.key)
         else:
             rep_type = self.accept(replacements)
-            dict_type = self.chk.named_generic_type('builtins.dict',
+            dict_type = self.chk.named_generic_type('typing.Mapping',
                                             [AnyType(), AnyType()])
```

Comment:
Also I believe the key type must match the type of the string on the left of the '%' (in Python 3) or must be either str or unicode (Python 2).